src/rust/engine/src/nodes.rs

```diff
@@ -1054,26 +1056,46 @@ impl Node for NodeKey {
     };
 
     let context2 = context.clone();
-    future::lazy(|| {
-      if let Some(span_id) = maybe_span_id {
-        set_parent_id(span_id);
-      }
-      match self {
-        NodeKey::DigestFile(n) => n.run(context).map(NodeResult::from).to_boxed(),
-        NodeKey::DownloadedFile(n) => n.run(context).map(NodeResult::from).to_boxed(),
-        NodeKey::MultiPlatformExecuteProcess(n) => n.run(context).map(NodeResult::from).to_boxed(),
-        NodeKey::ReadLink(n) => n.run(context).map(NodeResult::from).to_boxed(),
-        NodeKey::Scandir(n) => n.run(context).map(NodeResult::from).to_boxed(),
-        NodeKey::Select(n) => n.run(context).map(NodeResult::from).to_boxed(),
-        NodeKey::Snapshot(n) => n.run(context).map(NodeResult::from).to_boxed(),
-        NodeKey::Task(n) => n.run(context).map(NodeResult::from).to_boxed(),
+    future::lazy({
```

Comment:
This actually needs to be submitted onto the `io_pool`, or else it will still be occurring on whatever thread is running this chain of futures, and will block it.

Reply:
So even though the watch is explicitly spawned on the io_pool, it won't run there unless the future wrapping it is also run on the io_pool?